### PR TITLE
Index buffers reverted to dirty flag instead of timestamps

### PR DESCRIPTION
**Pull request by alexcristici**
The index buffers were recreated for each drawable even though 2 or more drawables used the same index buffer. 

We need to recreate index buffers only when the buffer data is dirty.
New index buffer will be shared through the `indexes` variable for all drawables that are using it.

This change also fixes the Android emulator crash. 

I also removed some unused variables: `indexBuffer`, `attributeBuffers` and `instanceBuffers`.

### CODE REVIEW

**Comment by louwers** [comment]
Could you provide some more context in the PR description?

**Comment by alexcristici** [comment]
> Could you provide some more context in the PR description?

Sure, not ready yet, I thought I created as a draft. I will change now.

**Comment by alexcristici** [comment]
Ready for review.

File: platform/android/MapLibreAndroid/src/cpp/map_renderer.cpp [hunk 1]
```diff
@@ -257,7 +257,7 @@ void MapRenderer::onSurfaceCreated(JNIEnv& env, const jni::Object<AndroidSurface
     // If we're running the emulator with the OpenGL backend, we're going to crash eventually,
     // unless we enable this mitigation.
     if (inEmulator()) {
-        renderer->enableAndroidEmulatorGoldfishMitigation(true);
```
Comment (alexcristici) [inline platform/android/MapLibreAndroid/src/cpp/map_renderer.cpp:260]: @mwilsnd For now I just disabled old workaround for Android crash. I would remove the code after you confirm is not crashing on your side too.

File: platform/android/MapLibreAndroid/src/cpp/map_renderer.cpp [hunk 2]
```diff
@@ -257,7 +257,7 @@ void MapRenderer::onSurfaceCreated(JNIEnv& env, const jni::Object<AndroidSurface
...
+        renderer->enableAndroidEmulatorGoldfishMitigation(false);
```
Comment (alexcristici) [inline platform/android/MapLibreAndroid/src/cpp/map_renderer.cpp:260]: @mwilsnd For now I just disabled old workaround for Android crash. I would remove the code after you confirm is not crashing on your side too.

**Comment by sjg-wdw** [comment]
@TimSylvester cool with you?

**Review by TimSylvester — APPROVED** [review]

**Review by mwilsnd — APPROVED** [review]
Looks to be resolved, nice find!

**Comment by alexcristici** [comment]
Removed workaround for Android emulator crash: https://github.com/maplibre/maplibre-native/pull/2858.

**Pull request merged**